### PR TITLE
[ui][new] 3scale discovery annotation test

### PR DESCRIPTION
**Pull request by mcada**
//test: @3scale

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java [hunk 1]
```diff
@@ -654,4 +658,52 @@ private void waitForCallbackRedirect(String expectedPartOfUrl) {
             fail("Error while redirecting to " + expectedPartOfUrl, e);
         }
     }
+
+    @And("^set 3scale discovery variable to \"([^\"]*)\"")
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java]: and

File: ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java [hunk 2]
```diff
@@ -654,4 +658,52 @@ private void waitForCallbackRedirect(String expectedPartOfUrl) {
...
+    public void set3scaleEnvVar(String value) {
+        DeploymentConfig dc = OpenShiftUtils.getInstance().getDeploymentConfig("syndesis-server");
+
+        List<EnvVar> vars = dc.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+
+        Optional<EnvVar> exposeVia3SCALE = vars.stream().filter(a -> a.getName().equalsIgnoreCase("CONTROLLERS_EXPOSE_VIA3SCALE")).findFirst();
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java]: SCALE -> Scale?

File: ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java [hunk 3]
```diff
@@ -654,4 +658,52 @@ private void waitForCallbackRedirect(String expectedPartOfUrl) {
...
+
+        WebDriverRunner.getWebDriver().navigate().refresh();
+        TestUtils.sleepForJenkinsDelayIfHigher(5);
+    }
+
+    @Then("^check that 3scale annotations are present on integration \"([^\"]*)\"")
+    public void check3scaleAnnotations(String integrationName) {
+        Map<String, String> annotations = OpenShiftUtils.getInstance().getService(("i-" + integrationName).toLowerCase()).getMetadata().getAnnotations();
+
+        log.info(annotations.toString());
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java]: needed?

File: ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java [hunk 4]
```diff
@@ -654,4 +658,52 @@ private void waitForCallbackRedirect(String expectedPartOfUrl) {
...
+        assertThat(annotations)
+                .contains(entry("discovery.3scale.net/description-path", "/openapi.json"))
+                .contains(entry("discovery.3scale.net/port", "8080"))
+                .contains(entry("discovery.3scale.net/scheme", "http"));
+    }
+
+    //TODO: should be refactored after asmigala is done with api provider tests
+    @And("^select first api provider operation$")
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/CommonSteps.java]: And

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mcada** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**